From: antony at mysql dot com
Date: December 6 2006 8:27pm
Subject: bk commit into 5.1 tree (antony:1.2325) BUG#24270

Below is the list of changes that have just been committed into a local
5.1 repository of antony. When antony does a push these changes will
be propagated to the main repository and, within 24 hours after the
push, to the public repository.
For information on how to access the public repository
see http://dev.mysql.com/doc/mysql/en/installing-source-tree.html

ChangeSet@stripped, 2006-12-06 11:27:13-08:00, antony@stripped +1 -0
  Bug#24270
    "mysql_install_db doesn't work properly on 5.1"
    force default sql mode when in bootstrap. prevents entries in my.cnf
    from breaking the 'install'

  sql/mysqld.cc@stripped, 2006-12-06 11:27:10-08:00, antony@stripped +2 -0
    force default sql mode when in bootstrap. prevents entries in my.cnf
    from breaking the 'install'

# This is a BitKeeper patch.  What follows are the unified diffs for the
# set of deltas contained in the patch.  The rest of the patch, the part
# that BitKeeper cares about, is below these diffs.
# User:	antony
# Host:	ppcg5.local
# Root:	/Users/antony/Work/p2-bug24270.1

--- 1.583/sql/mysqld.cc	2006-12-06 11:27:21 -08:00
+++ 1.584/sql/mysqld.cc	2006-12-06 11:27:21 -08:00
@@ -3640,6 +3640,8 @@
 
   if (opt_bootstrap)
   {
+    global_system_variables.tx_isolation= ISO_REPEATABLE_READ;
+    global_system_variables.sql_mode= fix_sql_mode(0);
     select_thread_in_use= 0;                    // Allow 'kill' to work
     bootstrap(stdin);
     end_thr_alarm(1);				// Don't allow alarms
